yak_ros/include/yak_ros/online_fusion_server_ros1.h

```diff
@@ -116,6 +119,11 @@ class OnlineFusionServer
   Eigen::Affine3d tsdf_frame_to_camera_prev_;
   /** @brief TF frame associated with the TSDF volume. */
   std::string tsdf_frame_;
+  /** @brief Queue that stores images until their transforms are available. This allows for the mismatch in timing
+   * between images being published and TFs */
+  std::queue<sensor_msgs::ImageConstPtr> image_queue_;
```

Comment:
Would this be something `tf2_ros::MessageFilter` could be used for?

Reply:
The answer is "probably yes", though I think it can be handled through a new PR rather than a modification to this one.

I added #15 to track this.